### PR TITLE
RCIAM-142_Hide_Attribute_field_during_enrollment

### PR DESCRIPTION
**Pull request by ioigoume**
This feature requires a schema update:
`alter table cm_co_enrollment_attributes add column form_hidden boolean;`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 - Add `hidden` functionality to Enroll page. The Admin can enable the functionality by changing the value of `Hide Enrollment Flow` field to true, in the config page of an Enrollment flow. By default the value is false/empty and all the configured Enrolment Flows will be displayed in `People->Enroll` page.
 - Retrieve AuthenticatingAuthority and depict in CO Person's canvas/profile
 - IdP hinting for RCAUTH plugin
+- Visually hide Enrollment Flow attribute fields during enrollment flow process
```
Comment (NicolasLiampotis) [inline CHANGELOG.md]: `Add support for hiding attributes from the enrollment form. Admins can hide an enrollment flow attribute by setting the value of `Hidden from enrollment form view` to true in the Enrollment Attribute configuration page. By default the value is false/empty and all the configured Enrolment Attributes will be displayed in the enrollment form.`

File: app/Config/Schema/schema.xml [hunk 2]
```diff
@@ -1398,6 +1398,7 @@
     </field>
     <field name="revision" type="I" />
     <field name="deleted" type="L" />
+    <field name="form_hide" type="L" />
```
Comment (NicolasLiampotis) [inline app/Config/Schema/schema.xml]: Change to `form_hidden` to be consistent with the existing `hidden` column

File: app/Lib/lang.php [hunk 3]
```diff
@@ -1228,6 +1228,8 @@
   'fd.groups' =>      'Groups',
   'fd.hidden' =>      'Hidden',
   'fd.hidden.desc' => 'If true, this field will not be rendered during enrollment',
+  'fd.hidden.form' => 'Visually Hidden',
```
Comment (NicolasLiampotis) [inline app/Lib/lang.php]: Replace `Visually Hidden` with `Hidden from enrollment form view`

File: app/Model/CoEnrollmentAttribute.php [hunk 4]
```diff
@@ -88,6 +88,12 @@ class CoEnrollmentAttribute extends AppModel {
       'required' => false,
       'allowEmpty' => true
     ),
+    'form_hide' => array(
```
Comment (NicolasLiampotis) [inline app/Model/CoEnrollmentAttribute.php]: `form_hidden` as above

File: app/View/CoEnrollmentAttributes/fields.inc [hunk 5]
```diff
@@ -493,6 +493,19 @@
...
       </tr>
+      <tr class="line<?php print ($l % 2); $l++; ?>">
+        <td>
+          <b><?php print _txt('fd.hidden.form'); ?></b><br />
+          <span class="descr"><?php print _txt('fd.hidden.form.desc'); ?></span>
+        </td>
+        <td>
+          <?php // Note this is a CoEnrollmentAttribute field, not CoEnrollmentAttributeDefault
+          print ($e
+            ? $this->Form->input('form_hide', array('default' => false))
```
Comment (NicolasLiampotis) [inline app/View/CoEnrollmentAttributes/fields.inc]: `form_hidden`

File: app/View/CoEnrollmentAttributes/fields.inc [hunk 6]
```diff
@@ -493,6 +493,19 @@
...
+            : ($co_enrollment_attributes[0]['CoEnrollmentAttribute']['form_hide']
```
Comment (NicolasLiampotis) [inline app/View/CoEnrollmentAttributes/fields.inc]: `form_hidden`

File: app/View/CoPetitions/petition-attributes.inc [hunk 7]
```diff
@@ -204,8 +204,10 @@
     }
 
     $mvpa = (count($coe_attributes) > 1);
+    // Check if the field should be displayed. If not just hide the div
+    $model_box_display = $coe_attributes[0]['CoEnrollmentAttribute']['form_hide'] ? "none" : "block";
```
Comment (NicolasLiampotis) [inline app/View/CoPetitions/petition-attributes.inc]: `form_hidden`

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: app/View/CoEnrollmentAttributes/fields.inc (hunk 6 above)
Comment (ioigoume) [reply app/View/CoEnrollmentAttributes/fields.inc]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (ioigoume) [reply CHANGELOG.md]: Done

File: app/Config/Schema/schema.xml (hunk 2 above)
Comment (ioigoume) [reply app/Config/Schema/schema.xml]: Done

File: app/Lib/lang.php (hunk 3 above)
Comment (ioigoume) [reply app/Lib/lang.php]: Done

File: app/Model/CoEnrollmentAttribute.php (hunk 4 above)
Comment (ioigoume) [reply app/Model/CoEnrollmentAttribute.php]: done

File: app/View/CoEnrollmentAttributes/fields.inc (hunk 5 above)
Comment (ioigoume) [reply app/View/CoEnrollmentAttributes/fields.inc]: Done

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**